frontend/src/lib/k8s/cluster.ts

```diff
@@ -430,6 +437,30 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
       return this.apiEndpoint.isNamespaced;
     }
 
+    static get readOnlyFields(): JsonPath<T>[] {
```

Comment:
I was wrong about asking for this to be a getter, as overriding this in the subclasses, as you did, requires declaring the _readOnlyFields, which is not a good practice. Let's turn _readOnlyFields into readOnlyFields and make it public.